src/err/err_state.rs

```diff
@@ -45,9 +46,14 @@ where
 }
 
 pub(crate) fn boxed_args(
```

Comment:
Since this is never used to construct anything but a `PyErrState`, why not turn it into a constructor, e.g.

```rust
impl PyErrState {
  fn lazy(ptype: &PyAny, args: impl PyErrArguments + 'static) -> Self { .. }
}
```

Reply:
Good idea, will rework tomorrow